config/v1/types_oauth.go

```diff
@@ -117,6 +120,16 @@ type OAuthIdentityProvider struct {
 	// +optional
 	GrantMethod GrantHandlerType `json:"grantMethod"`
 
+	// unqualifiedGroups determines if groups asserted by this provider should be prepended.
+	// If false, all valid groups asserted by this provider are prepended with openshift:idp:groups:<idp_name>:
+	// If true, all valid groups are used as-is without any modification.
+	// A group name is considered invalid if it:
+	// 1. Is empty
+	// 2. Equals . or .. or ~
+	// 3. Contains / or % or :
+	// +optional
+	UnqualifiedGroups bool `json:"unqualifiedGroups,omitempty"`
```

Comment:
`UnqualifiedGroups` makes me think it's a list of groups that shouldn't be qualified, not that all groups shouldn't be qualified with the prefix. Not sure what exactly would be better, `NoQualifiedGroupPrefix` maybe?

Reply:
No strong opinion on this one, `unqualifiedGroups` is what Simo came up with.  It was better than whatever terrible name I had before that.

Reply:
I find usage of the negative confusing.  How about `QualifiedGroups` , if true, we prepend?

Reply:
linking to @enj's comment: https://github.com/openshift/api/pull/151#issuecomment-451255619

I'd still prefer something like `QualifiedGroupPrefixOff` or `OmitQualifiedGroupPrefix`.